src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,192 @@
+
+    private double blinkTime = 1;
+    private double fadeDuration = 1;
+    private int rainbowHue = 0;
+
+    private Color primary = new Color(0);
+    private Color secondary = new Color(0);
+    private Color currentColor = primary;
+
+    private final Timer timer = new Timer();
+    public Translation3d solution = new Translation3d();
+    private Mode mode = Mode.SOLID;
+
+    public LedStrip(int port, int length) {
+        ledStrip = new AddressableLED(port);
+        ledBuffer = new AddressableLEDBuffer(length);
+
+        ledStrip.setLength(length);
+        ledStrip.setData(ledBuffer);
+        ledStrip.start();
+
+        timer.start();
+        timer.reset();
+    }
+
+    /**
+     * Sets the length of the led strip.
+     *
+     * @param length Number of leds in the strip.
+     */
+    public void setLength(int length) {
+        ledBuffer = new AddressableLEDBuffer(length);
+        ledStrip.setLength(length);
+    }
+
+    /**
+     * Sets the primary color of the strip.
+     *
+     * @param color Primary color.
+     */
+    public void setPrimary(Color color) {
+        primary = color;
+    }
+
+    /**
+     * Sets the secondary color of the strip.
+     *
+     * @param color Secondary color.
+     */
+    public void setSecondary(Color color) {
+        secondary = color;
+    }
+
+    /**
+     * Switch mode to solid color.
+     */
+    public void solid() {
+        mode = Mode.SOLID;
+    }
+
+    /**
+     * Switch mode to blink.
+     * (Should use setBlinkTime before this)
+     */
+    public void blink() {
+        mode = Mode.BLINK;
+    }
+
+    /**
+     * Switch mode to fade.
+     * (Should use setFadeDuration before this)
+     */
+    public void fade() {
+        mode = Mode.FADE;
+    }
+
+    /**
+     * Switch mode to rainbow.
+     */
+    public void rainbow() {
+        mode = Mode.RAINBOW;
+    }
+
+    /**
+     * Sets the led blink time.
+     *
+     * @param blinkTime Blink time in seconds
+     */
+    public void setBlinkTime(double blinkTime) {
+        this.blinkTime = blinkTime;
+    }
+
+    /**
+     * Sets the duration of the fade effect.
+     *
+     * @param duration Fade duration in seconds.
+     */
+    public void setFadeDuration(double duration) {
+        fadeDuration = duration;
+    }
+
+    /**
+     * Sets the led strip to a solid color
+     *
+     * @param color Color to set the strip to.
+     */
+    private void setSolidColor(Color color) {
+        var appliedColor = new edu.wpi.first.wpilibj.util.Color(color.getRed(), color.getGreen(), color.getBlue());
+        for (int i = 0; i < ledBuffer.getLength(); i++) {
+            ledBuffer.setLED(i, appliedColor);
+        }
+        ledStrip.setData(ledBuffer);
+    }
+
+    /**
+     * Interpolates between two colors.
+     *
+     * @param initial Initial color.
+     * @param goal    Final Color.
+     * @return
+     */
+    private Translation3d colorInterpolation(Color initial, Color goal) {
```

Comment:
Could you please explain what "color interpolation" means?

Reply:
To get the fade effect I saved the two HSB values as 3d points and interpolated them to get the correct color for the fade effect at the current time.

Reply:
Do you perhaps have a video of that you could send me? I can't really imagine how the effect looks.

Reply:
I don't really have a video and there's still no robot to take a video of.
It starts at one color and smoothly transitions to a different one.

Reply:
OK, I think I understand it better now. I like the implementation! I'm not sure though how you plan on using this method, perhaps add a public function that actually uses this method.